### PR TITLE
[GNA] FQ accuracy fixes 

### PR DESCRIPTION
**Pull request by mryzhov**
### Details:
 - Improved accuracy after POT optimizations

### Tickets:
 - 61724

### CODE REVIEW

File: inference-engine/src/gna_plugin/transformations/swap_input_matmul_gna.cpp [hunk 1]
```diff
@@ -32,63 +85,89 @@ SwapInputMatMul::SwapInputMatMul() {
...
+    };
 
-        ngraph::NodeVector new_ops;
+    auto m = std::make_shared<ngraph::pattern::Matcher>(matmul, matcher_name);
+    this->register_matcher(m, callback);
+}
 
-        if (shape_input_a[0] < 8 || ((shape_input_a[0] % 8 != 0 || shape_input_a[1] % 8 != 0))) {
+SwapInputMatMulWithBias::SwapInputMatMulWithBias() {
+    MATCHER_SCOPE(SwapInputMatMul);
```
Comment (nkogteva) [inline inference-engine/src/gna_plugin/transformations/swap_input_matmul_gna.cpp:101]: `MATCHER_SCOPE(SwapInputMatMulWithBias)`

File: inference-engine/src/gna_plugin/transformations/swap_input_matmul_gna.cpp [hunk 2]
```diff
@@ -32,63 +85,89 @@ SwapInputMatMul::SwapInputMatMul() {
...
-            }
-        }
+    auto m = std::make_shared<ngraph::pattern::Matcher>(add, matcher_name);
+    this->register_matcher(m, callback);
+}
 
-        auto traspose_output = create_transpose(new_matmul,  matmul->get_friendly_name());
-        new_ops.push_back(traspose_output);
+SwapInputMatMulWithFq::SwapInputMatMulWithFq() {
+    MATCHER_SCOPE(SwapInputMatMul);
```
Comment (nkogteva) [inline inference-engine/src/gna_plugin/transformations/swap_input_matmul_gna.cpp:134]: `MATCHER_SCOPE(SwapInputMatMulWithFq)`

File: inference-engine/src/gna_plugin/transformations/swap_input_matmul_gna.hpp [hunk 3]
```diff
@@ -16,4 +16,16 @@ class SwapInputMatMul: public ngraph::pass::MatcherPass {
     NGRAPH_RTTI_DECLARATION;
     SwapInputMatMul();
 };
+
```
Comment (nkogteva) [inline inference-engine/src/gna_plugin/transformations/swap_input_matmul_gna.hpp:19]: At least transformation tests for new transformations are required

File: inference-engine/src/legacy_api/src/transformations/convert_opset1_to_legacy/fc_bias_fusion.cpp [hunk 4]
```diff
@@ -43,7 +43,9 @@ ngraph::pass::FullyConnectedBiasFusion::FullyConnectedBiasFusion() {
         Shape bias_shape(bias->get_shape());
         Shape output_shape(fc->get_shape());
         size_t bias_size = std::accumulate(bias_shape.begin(), bias_shape.end(), size_t{1}, std::multiplies<int64_t>());
-        if (bias_shape.empty() || bias_shape.back() != output_shape.back() || bias_shape.back() != bias_size) {
+        if (bias_shape.empty() ||
+            (bias_shape.back() != output_shape.back() && bias_shape.back() != 1) ||
```
Comment (nkogteva) [inline inference-engine/src/legacy_api/src/transformations/convert_opset1_to_legacy/fc_bias_fusion.cpp:47]: I don't fully understand this change. Could you please explain this case?

File: inference-engine/src/gna_plugin/transformations/swap_input_matmul_gna.hpp (hunk 3 above)
Comment (mryzhov) [reply inference-engine/src/gna_plugin/transformations/swap_input_matmul_gna.hpp]: removed those transformations, they needed to fix another issue adn added here by mistake

File: inference-engine/src/legacy_api/src/transformations/convert_opset1_to_legacy/fc_bias_fusion.cpp (hunk 4 above)
Comment (elilobanova) [reply inference-engine/src/legacy_api/src/transformations/convert_opset1_to_legacy/fc_bias_fusion.cpp:47]: In the case when the bias last dimension is 1 but fully connected layer last output dimension is not 1 broadcast should be applied for the bias but it is still should be fused with fully connected layer. Without this change fusion isn't done for the case where broadcast is required.

File: inference-engine/src/gna_plugin/transformations/swap_input_matmul_gna.cpp (hunk 2 above)
Comment (mryzhov) [reply inference-engine/src/gna_plugin/transformations/swap_input_matmul_gna.cpp]: excluded from PR

File: inference-engine/src/gna_plugin/transformations/swap_input_matmul_gna.cpp (hunk 1 above)
Comment (mryzhov) [reply inference-engine/src/gna_plugin/transformations/swap_input_matmul_gna.cpp]: excluded from PR

File: inference-engine/src/legacy_api/src/transformations/convert_opset1_to_legacy/fc_bias_fusion.cpp (hunk 4 above)
Comment (mryzhov) [reply inference-engine/src/legacy_api/src/transformations/convert_opset1_to_legacy/fc_bias_fusion.cpp:47]: Added corresponding test

**Review by elilobanova — APPROVED** [review]

**Pull request merged**